src/components/icons/AnimatedIcon.tsx

```diff
@@ -1,10 +1,22 @@
+import { useRef } from 'react'
+import { useSelector } from 'react-redux'
 import { css, cx } from '../../../styled-system/css'
 import { icon } from '../../../styled-system/recipes'
 import { token } from '../../../styled-system/tokens'
 import AnimatedIconType from '../../@types/AnimatedIconType'
 import { ICON_SCALING_FACTOR } from '../../constants'
+import theme from '../../selectors/theme'
 import LottieAnimation from './LottieAnimation'
 
+/** Function to Convert RGB Color to Hex. */
+const rgbToHex = (color: string): string => {
```

Comment:
You can use `rgbToHex` provided by `@mui/material` which is already included in the project.

Reply:
The custom rgbToHex function was successfully replaced with the @mui/material utility.